Jetpack E2E: don't set `variant: siteEditor` in feature keys if testing Jetpack deploys in `Site Editor: Smoke Test`.

worldomonation opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/wp-calypso/issues/80453.

## Proposed Changes

This PR adds override criterion to the existing list for FSE: Smoke Test spec.

Key changes:
- navigate directly to the FSE page for `ATOMIC_VARIATION=ecomm-plan`.
- when determining a test account to use, only add the `variant: siteEditor` clause if the test being run is for non-Jetpack E2E.

## Testing Instructions

Ensure the following build configurations are passing:
  - [ ] Jetpack E2E Atomic Deploy

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

worldomonation committed:
test/e2e/specs/fse/fse__temp-smoke-test.ts
- update the override feature for the ATOMIC_VARIANT=ecomm-plan.
- add workaround for sidebar interaction of the ecomm-plan user.

worldomonation committed:
test/e2e/specs/fse/fse__temp-smoke-test.ts
- add missing protocol
- add wait for ecomm-plan user

worldomonation committed:
test/e2e/specs/fse/fse__temp-smoke-test.ts
- fix how the feature keys are being generated

worldomonation committed:
test/e2e/specs/fse/fse__temp-smoke-test.ts
- change the if clause condition

worldomonation committed:
test/e2e/specs/fse/fse__temp-smoke-test.ts
- revise criteria.

worldomonation committed:
test/e2e/specs/fse/fse__temp-smoke-test.ts
- key off `JETPACK_TARGET` env var instead

worldomonation commented on test/e2e/specs/fse/fse__temp-smoke-test.ts:
```diff
@@ -23,14 +23,27 @@ declare const browser: Browser;
+	// no valid match because none of the Jetpack deploy test users in the default criteria have
+	// `variant` keys.
+	let features = envToFeatureKey( envVariables );
+	if ( envVariables.JETPACK_TARGET !== 'wpcom-deployment' ) {
+		features = { ...features, variant: 'siteEditor' };
```
I'm totally open to a better, more intuitive way to override the feature keys here! 

I spent some time trying to understand how `envToFeatureKey` and `getTestAccountByFeature` interact with the default criteria and it only kept getting more confusing.

Through experimenting I found the `variant: 'siteEditor'` addition to the feature key was interfering with the test account matching in Jetpack E2E scenario. I decided to key off the one environment variable that changes between "normal" Calypso runs and Jetpack runs.

@dpasque - I'm definitely open to better ideas here.

dpasque replied:
Ah yeah... The `variant` key can be tricky to work with, especially when we have these more niche users where we aren't targeting meaningful variations between block theme users and non-block theme users.

I think what you have here is fine! But, I think my preference would be to actually update the default users [here](https://github.com/Automattic/wp-calypso/blob/trunk/packages/calypso-e2e/src/lib/utils/criteria-for-test-accounts.ts#L118-L164) and just duplicate the entries but also adding the `variant: 'siteEditor'` key.

The way I see it, it's like the test script/environment should declare the conditions they need, and then we can satisfy those conditions how we see fit. We just happen to have only one set of Jetpack atomic users that we use regardless of whether the script asks for block theme users or not. So, I think we should just make the user criteria tree reflect that!

It's a bit janky though still haha... And what you have here effectively gets at the same thing, just in a different way. So I'm fine with either. 😄

worldomonation replied:
Gotcha. It did occur to me that I should upgrade the default matrix instead, and not sure why I didn't do that. Probably because I tried to stay away from the whole `getTestAccountByFeature` side of things.

dpasque reviewed (approved):
LGTM! I added one comment about a potential different way to handle the site variation, but that's take-it-or-leave-it. 🙂

worldomonation committed:
test/e2e/specs/fse/fse__temp-smoke-test.ts
- remove the clause.

packages/calypso-e2e/src/lib/utils/get-test-account-by-feature.ts
- duplicate the Jetpack Atomic deploy list and add "variant: siteEditor" to the list.

worldomonation committed:
test/e2e/specs/fse/fse__temp-smoke-test.ts
- update comment
